src/components/App.tsx

```diff
@@ -382,22 +400,47 @@ export class App extends React.Component<any, AppState> {
         });
       });
       this.socket.on("new-user", async (socketID: string) => {
-        this.broadcastSocketData({
-          type: "SCENE_UPDATE",
-          payload: {
-            elements: elements.filter(element => {
-              return element.id !== this.state.editingElement?.id;
-            }),
-            appState: this.state,
-          },
-        });
+        this.broadcastSocketData({ type: "SCENE_UPDATE" });
       });
     }
   };
 
-  private broadcastSocketData = async (data: SocketUpdateData) => {
+  private broadcastSocketData = async (data: SocketUpdateDataSource) => {
     if (this.socketInitialized && this.socket && this.roomID && this.roomKey) {
-      const json = JSON.stringify(data);
+      let socketData;
+      switch (data.type) {
```

Comment:
This still seems weird to me to create a temporary data structure just to switch on it instantly afterwards.

What about `broadcastSocketData` take a single argument of type `{type: string, payload: any}` and create two functions

`broadcastMouseLocation` and `broadcastSceneUpdate`.

Reply:
This is internal implem that's not exposed to users of this API. Besides, we'd need to transform the data anyway, and if we split it into two functions it won't solve anything. In fact we'll need to either duplicate the encrypting & socket broadcasting logic, or we'll need to factor it out (and thus end up with 3 functions instead of 1) -- so not sure how it'll improve anything.

Reply:
having one function to encrypt and send and two functions to prepare the data for 2 different use cases make the most sense to me. But, if you prefer to leave as is, that’s fine.

Reply:
Ok, I'll see what I can do.